pylops/waveeqprocessing/_kirchhoff_cuda.py

```diff
@@ -0,0 +1,605 @@
+                        aptap = (
+                            aptap
+                            * aperturetap[
+                                int(
+                                    20
+                                    * (abs_angle_rec - angleaperturemin)
+                                    // dangleaperture
+                                )
+                            ]
+                        )
+                    # identify x-index of image point
+                    iz = ii % nz
+                    # aperture check
+                    aperture = abs(sixisrcrec - rixisrcrec) / iz
+                    if aperture < aperturemax:
+                        if aperture >= aperturemin:
+                            # extract aperture taper value
+                            aptap = (
+                                aptap
+                                * aperturetap[
+                                    int(20 * ((aperture - aperturemin) // daperture))
+                                ]
+                            )
+                        # time limit check
+                        if 0 <= itravii < nt - 1:
+                            ind1 = ii
+                            val1 = (
+                                (
+                                    x[isrc * nr + irec, itravii] * (1 - travdii)
+                                    + x[isrc * nr + irec, itravii + 1] * travdii
+                                )
+                                * damp
+                                * aptap
+                            )
+                            cuda.atomic.add(y, ind1, val1)
+
+    def _matvec_call(self, *inputs):
+        """Handle the forward operation call, dispatching to appropriate CUDA kernels.
+
+        This method selects the appropriate kernel to execute based on the computation flags,
+        and performs the forward operation.
+
+        Parameters
+        ----------
+        *inputs : :obj:`list`
+            List of input parameters required by the kernels.
+
+        Returns
+        -------
+        y : :obj:`numpy.ndarray`
+            Output data (seismic data) of forward operation.
+
+        """
+        if self.dynamic:
+            y_d = self._call_dynamic(inputs[0], "_matvec")
+        else:
+            y_d = self._call_nondynamic("_matvec", *inputs)
+        """
+        else:
+            x_d = inputs[0]
+            y_d = inputs[1]
+            ns_d = np.int32(inputs[2])
+            nr_d = np.int32(inputs[3])
+            nt_d = np.int32(inputs[4])
+            ni_d = np.int32(inputs[5])
+            dt_d = np.float32(inputs[6])
+            trav_srcs_d = to_cupy(inputs[7])
+            trav_recs_d = to_cupy(inputs[8])
+            self._travsrcrec_kirch_matvec_cuda[
+                self.num_blocks, self.num_threads_per_blocks
+            ](x_d, y_d, ns_d, nr_d, nt_d, ni_d, dt_d, trav_srcs_d, trav_recs_d)
+            cuda.synchronize()
+        """
```

Comment:
Use comment, not string

Reply:
Sorry, that's my bad 😄 I forgot to take it away after I moved these to a function...

Reply:
Fixed